src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobExecutionService.java

```diff
@@ -398,14 +397,18 @@ private void processTestCaseNode(XMLStreamReader xmlStreamReader, List<LocalCIBu
         // Check if there is a failure node inside the testcase node.
         // Call next() until there is an end element (no failure node exists inside the testcase node) or a start element (failure node exists inside the
         // testcase node).
-        xmlStreamReader.next();
-        while (!(xmlStreamReader.isEndElement() || xmlStreamReader.isStartElement())) {
+        do {
             xmlStreamReader.next();
         }
+        while (!(xmlStreamReader.isEndElement() || xmlStreamReader.isStartElement()));
         if (xmlStreamReader.isStartElement() && "failure".equals(xmlStreamReader.getLocalName())) {
```

Comment:
Tests can also fail with `error` nodes whose text should be extracted, see https://github.com/testmoapp/junitxml/blob/main/examples%2Fjunit-complete.xml#L109. Is this covered anywhere?